### DISCUSSION TITLE
Arbitrary values

### DISCUSSION THREAD

**Discussion by shaunforage** [joe-bell/cva#218] [General]
Love the work on cva so far. 

Just wondering if there is a way to use arbitrary values with [variants](https://tailwindcss.com/docs/adding-custom-styles#using-arbitrary-values). Basically I want a variant with some set options while allowing a rem value also. I am imagining a syntax where you can use the value within the classname.

![Screen Shot 2023-09-13 at 10 47 21 am](https://github.com/joe-bell/cva/assets/106948097/4369e39d-04c0-4cfd-bbf8-a917c79129ba)

**Comment by joe-bell** [comment]
Sadly not really possible :(

Tailwind’s JIT engine doesn’t support dynamic strings. Exploring some documentation-based approaches to work around this though

**Reply by shaunforage** [reply]
Thanks for the reply 🙏 Looking forward to the future for cva!

**Comment by ivosabev** [comment]
It could work if you add the classes to the `safelist`. 

https://tailwindcss.com/docs/content-configuration#safelisting-classes

For the convenience dynamic values offers, I would go through the burden to manually whitelist the needed values.

**Reply by joe-bell** [reply]
Absolutely you could, and it's something I've been exploring. The key issue is finding a way to extract all of these classes out prior to Tailwind's compile step – which would be significant overhead
